### PR TITLE
feat: added report api

### PR DESCRIPTION
**Pull request by zombozo12**
Added /reports API client based on this docs https://developers.xendit.co/api-reference/#reports

Related issue:  #64

### CODE REVIEW

File: report.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+type Report struct {
+	ID         string     `json:"id"`
+	Type       string     `json:"type"`
+	Filter     Filter     `json:"filter"`
+	Format     string     `json:"format"`
+	Status     string     `json:"status"`
+	Url        string     `json:"url"`
+	Currency   string     `json:"currency"`
+	BusinessID string     `json:"business_id"`
+	Created    *time.Time `json:"created"`
```
Comment (adityarx) [inline report.go:14]: I believe these (`Created` & `Updated`) are both non-nullable required parameters. Could we change the type from `*time.Time` -> `time.Time`

**Comment by adityarx** [comment]
Hello @zombozo12, thanks so much for taking the time to contribute to our library! I've left a few minor comments on the PR. I'll also ask an engineer from the Reports team to have a look at your PR, so hang tight! Thanks again 🙏

File: report/client.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
+package report
```
[the remark this answers is not in the document]
Comment (zombozo12) [reply report/client.go:1]: Ah, yes, I missed this one. Will add soon.

File: report.go (hunk 1 above)
Comment (zombozo12) [reply report.go:14]: Sure, didn't realize it was non-nullable. Will update it soon.

**Comment by zombozo12** [comment]
Hi @adityarx, thank you for your reviews. I've fixed all the issues, thank you! :pray:

**Review by ahmadhp — APPROVED** [review]

**Review by sekaranglila — APPROVED** [review]

**Comment by adityarx** [comment]
Hello @zombozo12, this has been released as version v1.019. Thanks so much for your contribution! 🙏

**Comment by zombozo12** [comment]
Hi @adityarx thank you so much for your guidance! 🙏

**Pull request merged**